### DISCUSSION TITLE
[FEATURE REQ] Seperate Release package for Linux (64) Only

### DISCUSSION THREAD

**Discussion by CoolZeroNL** [grafana/grafana-infinity-datasource#792] [Ideas]
I'm using the plugin within a docker image. This docker images im running on a Raspberry Pi (5) and running on a RockyOS vm. This means different Arch, PI needs ARM64 and RockOS need AMD64.

I build the docker images with a copy the release of the plugin release zip files.

However there is no option to only have the amd64 and arm64 executable within the plugin folder because of the signature validation.

As example, i placed the 2 releases in the grafana folder.
- /plugins/yesoreyeram-infinity-datasource-amd64/
- /plugins/yesoreyeram-infinity-datasource-arm64/

Grafana loads the plugins trough alphabetical so to see, because here the amd64 folder is loaded
```log
logger=plugins.registration t=2024-03-13T06:18:40.051132119Z level=info msg="Plugin registered" pluginId=yesoreyeram-infinity-datasource
```

Here u can see that it needs the `gpx_infinity_linux_arm64` but looks for it in `yesoreyeram-infinity-datasource-amd64` folder
```log
logger=plugins.backend.start t=2024-03-13T06:18:40.051562344Z level=error msg="Could not start plugin" pluginId=yesoreyeram-infinity-datasource error="fork/exec /opt/grafana/plugins/yesoreyeram-infinity-datasource-amd64/gpx_infinity_linux_arm64: no such file or directory"
logger=plugins.initialization t=2024-03-13T06:18:40.051592251Z level=error msg="Could not initialize plugin" pluginId=yesoreyeram-infinity-datasource error="fork/exec /opt/grafana/plugins/yesoreyeram-infinity-datasource-amd64/gpx_infinity_linux_arm64: no such file or directory"
```

And grafana cant load the `yesoreyeram-infinity-datasource-arm64` folder because it sees it as a duplicate..
```log
logger=plugins.dedupe t=2024-03-13T06:18:40.052031957Z level=warn msg="Skipping loading of plugin as it's a duplicate" pluginId=yesoreyeram-infinity-datasource
```

If i place the `gpx_infinity_linux_arm64` file into the `yesoreyeram-infinity-datasource-amd64` folder its compaining of the signature, what is logical

```log
logger=plugins.signature t=2024-03-13T06:27:17.297955532Z level=warn msg="The following files were not included in the signature" plugin=yesoreyeram-infinity-datasource files=[gpx_infinity_linux_arm64]
logger=plugins.signature.validation t=2024-03-13T06:27:17.298047922Z level=warn msg="Skipping loading plugin due to problem with signature" pluginId=yesoreyeram-infinity-datasource status=modified
logger=plugins.validation t=2024-03-13T06:27:17.298081978Z level=error msg="Plugin validation failed" pluginId=yesoreyeram-infinity-datasource error="plugin 'yesoreyeram-infinity-datasource' has an modified signature"

logger=plugins.registration t=2024-03-13T06:27:17.429156692Z level=info msg="Plugin registered" pluginId=yesoreyeram-infinity-datasource

logger=plugins.backend.start t=2024-03-13T06:27:17.429601157Z level=error msg="Could not start plugin" pluginId=yesoreyeram-infinity-datasource error="fork/exec /opt/grafana/plugins/yesoreyeram-infinity-datasource-arm64/gpx_infinity_linux_arm64: permission denied"
logger=plugins.initialization t=2024-03-13T06:27:17.429643231Z level=error msg="Could not initialize plugin" pluginId=yesoreyeram-infinity-datasource error="fork/exec /opt/grafana/plugins/yesoreyeram-infinity-datasource-arm64/gpx_infinity_linux_arm64: permission denied"
```

I could offcouse include the `yesoreyeram-infinity-datasource-2.5.0.zip` that contain all architecture executables, however this is unziped 250mb, and waiting space/docker size with unneeded executables.

My request is that:
  - could there be a linux(64) package release? 
    
    or

  - can each arch of the plugin, be a seperated signature plugin and can be loaded next to each other ?

**Comment by jwavoet** [comment]
Ran into the same issue when switching from amd64 to arm64 architecture. To solve it you need to:
1. Go into the container and remove the folder (including files) `/var/lib/grafana/plugins/yesoreyeram-infinity-datasource`. 
1. Download the correct files for arm64, example: `curl -L -o /tmp/infinity-3.6.0.zip "https://github.com/grafana/grafana-infinity-datasource/releases/download/v3.6.0/yesoreyeram-infinity-datasource-3.6.0.linux_arm64.zip"`. 
1. Extract `unzip /tmp/infinity-3.6.0.zip -d /var/lib/grafana/plugins/`. 
1. Restart Grafana and it should pickup the new arm64 plugin.

This only happens, it seems, when you switch architectures while keeping your plugin data.
